### PR TITLE
Feature/gh 1370 walletconnect scanner screen adjustments

### PR DESCRIPTION
**Pull request by MouazAlzahabi**
GH-1370 WalletConnect Scanner Screen Adjustments

### CODE REVIEW

File: SafeUIKit/SafeUIKit/QRCodeScanner/ScanQRCodeHandler.swift [hunk 1]
```diff
@@ -24,7 +24,7 @@ class ScanQRCodeHandler {
     private var scannerController: ScannerViewController?
     private var debugButtons = [DebugButton]()
     private var didFinishScanning = false
-
+    var header: String?
```
Comment (DmitryBespalov) [inline SafeUIKit/SafeUIKit/QRCodeScanner/ScanQRCodeHandler.swift:27]: please add a blank line between variable and enum declaration (after this line)

File: SafeUIKit/SafeUIKit/QRCodeScanner/ScannerViewController.swift [hunk 2]
```diff
@@ -24,17 +24,20 @@ class ScannerViewController: UIViewController {
...
+    @IBOutlet weak var headerLabel: UILabel!
     @IBOutlet weak var debugButtonsStackView: UIStackView!
     @IBOutlet weak var closeButton: UIButton!
     @IBOutlet weak var buttonEffectView: UIVisualEffectView!
 
     private var debugButtonReturnCodes = [String]()
     private var debugButtons = [UIButton]()
     private var shouldStopHandling: Bool = false
-
+    var header: String?
```
Comment (DmitryBespalov) [inline SafeUIKit/SafeUIKit/QRCodeScanner/ScannerViewController.swift:36]: please add a blank line after this line

File: SafeUIKit/SafeUIKit/QRCodeScanner/ScannerViewController.swift [hunk 3]
```diff
@@ -71,9 +74,11 @@ class ScannerViewController: UIViewController {
         codeReaderVC.view.frame = view.frame
         view.insertSubview(codeReaderVC.view, at: 0)
         codeReaderVC.didMove(toParent: self)
-
+
```
Comment (DmitryBespalov) [inline SafeUIKit/SafeUIKit/QRCodeScanner/ScannerViewController.swift]: please remove trailing whitespace

File: SafeUIKit/SafeUIKit/QRCodeScanner/ScannerViewController.swift [hunk 4]
```diff
@@ -144,5 +149,4 @@ fileprivate class CircleMask: BaseCustomView {
         shapeLayer.path = path
         shapeLayer.fillColor = ColorName.black.color.cgColor
     }
-
```
Comment (DmitryBespalov) [inline SafeUIKit/SafeUIKit/QRCodeScanner/ScannerViewController.swift]: please add the blank line back for consistency

File: SafeUIKit/SafeUIKit/QRCodeScanner/ScannerViewController.xib [hunk 5]
```diff
@@ -59,20 +74,74 @@
...
+                    <color key="backgroundColor" white="0.0" alpha="0.0" colorSpace="custom" customColorSpace="genericGamma22GrayColorSpace"/>
+                    <color key="tintColor" white="1" alpha="1" colorSpace="custom" customColorSpace="genericGamma22GrayColorSpace"/>
+                    <constraints>
+                        <constraint firstAttribute="width" secondItem="7gV-hi-CV2" secondAttribute="height" id="MdA-ip-sgM"/>
+                    </constraints>
+                </imageView>
             </subviews>
             <color key="backgroundColor" red="1" green="1" blue="1" alpha="1" colorSpace="custom" customColorSpace="sRGB"/>
             <constraints>
+                <constraint firstItem="AdF-AR-sr9" firstAttribute="width" secondItem="i5M-Pr-FkT" secondAttribute="width" id="1Wj-Gy-xne"/>
```
Comment (DmitryBespalov) [inline SafeUIKit/SafeUIKit/QRCodeScanner/ScannerViewController.xib:109]: The frame will look nicer if we add a 3pt inset of the overlay inside the frame.

Please make the constraints of the dimming views to overlap the "frame" image with 3pt

File: SafeUIKit/SafeUIKit/QRCodeScanner/ScannerViewController.swift [hunk 6]
```diff
@@ -71,9 +74,11 @@ class ScannerViewController: UIViewController {
         codeReaderVC.view.frame = view.frame
         view.insertSubview(codeReaderVC.view, at: 0)
         codeReaderVC.didMove(toParent: self)
-
+        
         closeButton.accessibilityLabel = LocalizedString("close", comment: "Close button on camera")
         buttonEffectView.mask = CircleMask(frame: buttonEffectView.bounds)
+        headerLabel.text = header
+        headerLabelContainerView.isHidden = header == nil
     }
```
Comment (DmitryBespalov) [inline SafeUIKit/SafeUIKit/QRCodeScanner/ScannerViewController.swift:84]: Small improvement: let's make the status bar white, so it is visible on the dark dimmed background.

```
    override var preferredStatusBarStyle: UIStatusBarStyle {
        .lightContent
    }
```

File: SafeUIKit/SafeAppUI/MainFlow/Menu/WalletConnect/WCSessionListTableViewController.swift [hunk 7]
```diff
@@ -65,6 +65,7 @@ import MultisigWalletApplication
             guard code.starts(with: "wc:") else { return nil }
             return code
         }
+        scanButtonItem.scanHeader = LocalizedString("scan_wallet_connect", comment: "Scan WalletConnect QR code")
```
Comment (sche) [inline SafeUIKit/SafeAppUI/MainFlow/Menu/WalletConnect/WCSessionListTableViewController.swift]: please place LocalizedString in enum Strings

**Review by DmitryBespalov — CHANGES_REQUESTED** [review]
Good 👍. Almost done - some small polishing left for it to be ready.

File: SafeUIKit/SafeUIKit/Components/QRCodeInput/QRCodeInput.swift [hunk 8]
```diff
@@ -21,7 +21,14 @@ public final class QRCodeInput: UITextField {
         }
     }
     var scanHandler = ScanQRCodeHandler()
-
+    public var scanHeader: String? {
```
Comment (sche) [inline SafeUIKit/SafeUIKit/Components/QRCodeInput/QRCodeInput.swift:24]: I am not sure if we use this component at all. Probably we just should delete it.

File: SafeUIKit/SafeUIKit/QRCodeScanner/ScannerViewController.swift (hunk 3 above)
Comment (sche) [inline SafeUIKit/SafeUIKit/QRCodeScanner/ScannerViewController.swift]: does linter show warning on this line now?

File: SafeUIKit/SafeUIKit/Components/QRCodeInput/QRCodeInput.swift (hunk 8 above)
Comment (MouazAlzahabi) [reply SafeUIKit/SafeUIKit/Components/QRCodeInput/QRCodeInput.swift:24]: True, It's not used so far but I added it for consistency as I did for all components that use QRScanner.

**Review by DmitryBespalov — APPROVED** [review]

**Comment by MouazAlzahabi** [comment]
closes #1370

**Pull request merged**